tests/integration/test_reads.py

```diff
@@ -506,3 +508,40 @@ def test_hive_locking(session_catalog_hive: HiveCatalog) -> None:
                 table.transaction().set_properties(lock="fail").commit_transaction()
         finally:
             open_client.unlock(UnlockRequest(lock.lockid))
+
+
+@pytest.mark.integration
+def test_hive_locking_with_retry(session_catalog_hive: HiveCatalog) -> None:
+    table = create_table(session_catalog_hive)
+    database_name: str
+    table_name: str
+    _, database_name, table_name = table.identifier
+
+    hive_client: _HiveClient = _HiveClient(session_catalog_hive.properties["uri"])
+
+    executor = ExecutorFactory.get_or_create()
+
+    with hive_client as open_client:
+
+        def another_task() -> None:
+            lock1: LockResponse = open_client.lock(session_catalog_hive._create_lock_request(database_name, table_name))
+            time.sleep(5)
```

Comment:
nit: `time.sleep` in test is typically an anti-pattern, this will add at least 5 seconds to the test suite in the future.

Reply:
It might be easier to mock `lock` and `check_lock` functions instead of relying on the timing of the function calls

Reply:
Also maybe add a test case for when `_wait_for_lock` failed to acquire locks after retry

Reply:
https://stackoverflow.com/questions/47906671/python-retry-with-tenacity-disable-wait-for-unittest

This might be helpful to override the waiting behavior in retry

Reply:
I have added a new unit test  `test_hive_wait_for_lock` in test_hive.py that uses mocked  `lock` and `check_lock`.

But I still keep an integration test based on the real hive metastore to simulate real-world cases.
In order to reduce the test latency, I use fine-grained sleep time instead.

WDYT?